packages/deployer/subtasks/validate-interfaces.js

```diff
@@ -0,0 +1,32 @@
+const { subtask } = require('hardhat/config');
+const mapValues = require('just-map-values');
+const logger = require('@synthetixio/core-js/utils/io/logger');
+const InterfaceCoverageASTValidator = require('../internal/interface-ast-validator');
+const { ContractValidationError } = require('../internal/errors');
+const { SUBTASK_VALIDATE_INTERFACES } = require('../task-names');
+
+subtask(SUBTASK_VALIDATE_INTERFACES).setAction(async (_, hre) => {
+  logger.subtitle('Validating all visible functions are defined in interfaces');
+
+  const { deployment } = hre.deployer;
+
+  const asts = mapValues(deployment.sources, (val) => val.ast);
+  const validator = new InterfaceCoverageASTValidator(asts);
```

Comment:
The validator takes an object of AST nodes, but on the constructor it uses is with `Object.values(asts)`. And here, we use `mapValues`.

I think we could just do `Object.values(deployment.sources).map((val) => val.ast)` here, and remove the `Object.values(asts)` from inside the validator.

Reply:
done